fix(BFormFloatingLabel): Clean up props

dwgray opened the pull request:
# Describe the PR

- The `text` prop on `BFormFloatingLabel` used the value of that property as the fallback content of the `default` slot. This doesn't work, that slot has to contain one of select few form controls. Putting plain text there just generates ugliness.
- Make slots required where appropriate (`BForm`, `BFormFloatingLabel` and `BFormRow`)

## Small replication

[stackblitz](https://stackblitz.com/edit/github-llmgfm?file=src%2Fcomponents%2FComp.vue)

```
<template>
  <Container>
    <BRow
      ><BCol>
        <BForm class="floating-label">
          <BInput
            id="floatingEmail"
            type="email"
            class="form-control"
            placeholder="Email address"
            label="Email address"
          />
          <BInput
            id="floatingPassword"
            type="password"
            placeholder="Password"
            label="Password"
          />
        </BForm> </BCol
    ></BRow>
    <BRow
      ><BCol>
        <BForm>
          <BFormFloatingLabel
            label="Email address"
            label-for="floatingEmail"
            class="my-2"
            text="Testing floating label text"
          >
          </BFormFloatingLabel>
        </BForm> </BCol
    ></BRow>
  </Container>
</template>
```
## PR checklist

<!-- (Update "[ ]" to "[x]" to check a box) -->

**What kind of change does this PR introduce?** (check at least one)

- [x] Bugfix :bug: - `fix(...)`
- [ ] Feature - `feat(...)`
- [ ] ARIA accessibility - `fix(...)`
- [ ] Documentation update - `docs(...)`
- [ ] Other (please describe)

**The PR fulfills these requirements:**

- [x] Pull request title and all commits follow the [**Conventional Commits**](https://www.conventionalcommits.org/) convention or has an [**override**](https://github.com/googleapis/release-please#how-can-i-fix-release-notes) in this pull request body **This is very important, as the `CHANGELOG` is generated from these messages, and determines the next version type. Pull requests that do not follow conventional commits or do not have an override will be denied**

dwgray commented on packages/bootstrap-vue-next/src/components/BForm/BForm.vue:
```diff
@@ -16,7 +16,6 @@ import type {BFormProps} from '../../types'
 import {computed, ref} from 'vue'
 
 const _props = withDefaults(defineProps<BFormProps>(), {
-  floating: false,
```
I have notes in the PR description and an example as to why I believe this property should be removed.

dwgray commented on packages/bootstrap-vue-next/src/components/BForm/BFormFloatingLabel.vue:
```diff
@@ -24,7 +22,7 @@ const props = useDefaults(_props, 'BFormFloatingLabel')
 
 defineSlots<{
   // eslint-disable-next-line @typescript-eslint/no-explicit-any
-  default?: (props: Record<string, never>) => any
```
I have notes in the PR description and an example as to why I believe this property should be removed.

VividLemon commented:
Not sure why you think you can't have form-floating on a form element
![image](https://github.com/bootstrap-vue-next/bootstrap-vue-next/assets/31498096/56fd4a18-08a6-46ce-86f5-6f4a3252464e)

Idrk why one would want to use form over a div in this case though

dwgray commented:
I think those examples in BS5 are misleading. They only work if you have a single control/label combination in the form.  See [this stackblitz](https://stackblitz.com/edit/aqveji?file=index.html) for a counter-example in native BS5.

Us putting a prop called `floating` implies to me that all of the controls in the form will somehow be floating (maybe we could implement that, but it would be complicated and is not accomplished by merely putting `form-floating` on the form element. Someone who wanted to mimic the single control example could add the class on the `BForm` themselves and would hopefully know enough not to shoot themselves in the foot.

If you really want to keep that prop in, let me know.  In that case, I'd also like to add some stuff to the docs so people don't assume it does more than it does. Alternately, I could take it out and include a single form control example. Let me know which you prefer.

VividLemon commented:
Regardless on if the class is misleading, it is native bootstrap form code. There's no reason to remove it

VividLemon commented on packages/bootstrap-vue-next/src/components/BForm/BForm.vue:
```diff
@@ -35,7 +34,6 @@ defineSlots<{
 }>()
 
 const computedClasses = computed(() => ({
-  'form-floating': props.floating,
```
This is probably my only issue with this PR. I suppose some components could benefit from making their slots not optional as shwon in BFormFloatingLabel. But this could be something else, or not added

VividLemon commented on packages/bootstrap-vue-next/src/components/BForm/form.spec.ts:
```diff
@@ -28,15 +28,6 @@ describe('form', () => {
     expect(wrapper.attributes('novalidate')).toBeUndefined()
   })
 
```
```suggestion

it('has class form-floating when prop floating', async () => {
    const wrapper = mount(BForm, {
      props: {floating: true},
    })
    expect(wrapper.classes()).toContain('form-floating')
    await wrapper.setProps({floating: false})
    expect(wrapper.classes()).not.toContain('form-floating')
  })

```

VividLemon commented on packages/bootstrap-vue-next/src/components/BForm/form.spec.ts:
```diff
@@ -28,7 +28,7 @@ describe('form', () => {
   })
 
-  it('has class form-floating when prop floating', async () => {
+it('has class form-floating when prop floating', async () => {
```
```suggestion
  it('has class form-floating when prop floating', async () => {
```

dwgray commented:
> Regardless on if the class is misleading, it is native bootstrap form code. There's no reason to remove it

Okay, I will defer to you on this. Added that property back in. I have #2003 that was in progress when I saw this that fixes some other docs issues. I've added some documentation about the `floating` label that I hope will make its limited value clear to users.
